internal/image/image.go

```diff
@@ -5,15 +5,17 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strings"
 
 	"github.com/docker/app/internal/slices"
 	composetypes "github.com/docker/cli/cli/compose/types"
+	"github.com/docker/distribution/reference"
 	"github.com/pkg/errors"
 )
 
 // Add add service images to the app package
 func Add(appname string, services []string, config *composetypes.Config) error {
-	if err := os.Mkdir(filepath.Join(appname, "images"), 0755); err != nil {
+	if err := os.Mkdir(filepath.Join(appname, "images"), 0755); err != nil && !os.IsExist(err) {
```

Comment:
`os.MkdirAll` does that, isn't it ?

Reply:
Not exactly, it creates the whole directory tree which is not the behavior I want here.

Reply:
I agree with @mnottale here. It's clearer to do `Mkdir` and check if it exists than do a `MkdirAll`. In general I'd prefer that we only use `MkdirAll` if we absolutely have to.